config/abilities/provider_any.rb

```diff
@@ -21,7 +21,9 @@
     cannot %i[destroy update_role], user
 
     # Services
-    can %i[read show edit update], Service, user.accessible_services.where_values_hash
+    user_accessible_services = user.accessible_services
+    can %i[read show edit update], Service, user_accessible_services.where_values_hash unless user_accessible_services.is_a? ActiveRecord::NullRelation
```

Comment:
`user_accessible_services.is_a? ActiveRecord::NullRelation` was added to support `next none` line in the Service's `permitted_for` scope.

Otherwise, in case "no services are allowed", while `user.accessible_services` returns an empty array, `user.accessible_services.where_values_hash` is `{}`, which makes cancancan authorize ALL services (because there is no query conditions).

Adding this `unless` condition makes it possible to prohibit all services if `permitted_for` returns `none`

Reply:
```suggestion
    can %i[read show edit update], Service, user_accessible_services.presence&.where_values_hash
```